### ISSUE TITLE
logout failed : provide a valid authorization token

### ISSUE THREAD

**Issue by prashantjeena** [cosmic-byte/flask-restplus-boilerplate#14]
![Screenshot (1)](https://user-images.githubusercontent.com/50395643/66024410-6ac4d500-e511-11e9-9130-6a8b4870147e.png)

I reached at this point in my app, but when i tried executing login and logout in localhost login worked fine but logout gave this error :  provide a valid authorization token .

![Screenshot (2)](https://user-images.githubusercontent.com/50395643/66024610-e757b380-e511-11e9-97d7-cad8bd7a612f.png)

Please take a look. Thanks in advance

**Comment by ItStark** [comment]
could you solve that? @prashantjeena

**Comment by ItStark** [comment]
Hey, I've just solve the logout problem. You should specift an authorization dictionary and give it to Api as a parameter. For that:

Firstly add this ->

`authorizations = {
    'Basic Auth': {
        'type': 'apiKey',
        'in': 'header',
        'name': 'Authorization'
    },
}`

to the root directory __init__.py and then add **authorizations** dictionary as an **Api** param ->

`api = Api(blueprint,
          doc='/documentation',
          title='title here',
          version='1.0',
          description='a description',
          security='Basic Auth',
          authorizations=authorizations
          )`

This makes all api methods required auth. Last thing you should do is replaceing `auth_token = data.split(" ")[1]` with this: `auth_token = data.split(" ")[0]` in **logout_user** function in **auth_helper.py**, in first if block.

This worked for me, hope this helps you too.

**Comment by prashantjeena** [comment]
> Hey, I've just solve the logout problem. You should specify an authorization dictionary and give it to Api as a parameter. For that:
> 
> Firstly add this ->
> 
> `authorizations = { 'Basic Auth': { 'type': 'apiKey', 'in': 'header', 'name': 'Authorization' }, }`
> 
> to the root directory **init**.py and then add **authorizations** dictionary as an **Api** param ->
> 
> `api = Api(blueprint, doc='/documentation', title='title here', version='1.0', description='a description', security='Basic Auth', authorizations=authorizations )`
> 
> This makes all api methods required auth. Last thing you should do is replaceing `auth_token = data.split(" ")[1]` with this: `auth_token = data.split(" ")[0]` in **logout_user** function in **auth_helper.py**, in first if block.
> 
> This worked for me, hope this helps you too.

where to add the api line ?
when i added it to the same init file ....it said name Api is not defined.

**Comment by prashantjeena** [comment]
this is my init.py file
`from flask import Flask
from flask_sqlalchemy import SQLAlchemy
from flask_bcrypt import Bcrypt

from .config import config_by_name

db = SQLAlchemy()
flask_bcrypt = Bcrypt()

authorizations = { 'Basic Auth': { 'type': 'apiKey', 'in': 'header', 'name': 'Authorization' }, }
api = Api(blueprint, doc='/documentation', title='title here', version='1.0', description='a description', security='Basic Auth', authorizations=authorizations )
def create_app(config_name):
    app = Flask(__name__)
    app.config.from_object(config_by_name[config_name])
    db.init_app(app)
    flask_bcrypt.init_app(app)
    

    return app`

**Comment by ItStark** [comment]
No, not into this __init__.py file, you should put `authorizations` into other __init__.py (located in upper package)

There is two file named __init__.py. One is belong to **main** package and other one is belong to top package. My name of top package op **main** is **app**. So i put that into **app's** __init__.py.

You should do the same. Not into inner __init__.py

**Comment by prashantjeena** [comment]
hey, thanks for the response .......i tried the same .....this is my init.py in the root directory.........am i doing anything wrong in particular coz it says name 'app' not defined.
`
from flask import Flask
from flask_sqlalchemy import SQLAlchemy
from flask_bcrypt import Bcrypt

from .config import config_by_name

db = SQLAlchemy()
flask_bcrypt = Bcrypt()
authorizations = { 'Basic Auth': { 'type': 'apiKey', 'in': 'header', 'name': 'Authorization' }, }
api = app.Api(blueprint, doc='/documentation', title='title here', version='1.0', description='a description', security='Basic Auth', authorizations=authorizations )

def create_app(config_name):
    app = Flask(__name__)
    app.config.from_object(config_by_name[config_name])
    db.init_app(app)
    flask_bcrypt.init_app(app)
    return app`

**Comment by ma3achou** [comment]
@prashantjeena !! but you have to set the token in your Header request, see the screenshot

![image](https://user-images.githubusercontent.com/849601/82637691-4ff3bd80-9bd3-11ea-954a-6bb763f3b41e.png)

**Comment by Kpatric** [comment]
Getting this error on auth_helper logout_user method   
auth_token = data.split("")[0]
AttributeError: 'dict' object has no attribute 'split'

**Comment by ma3achou** [comment]
> Getting this error on auth_helper logout_user method
> auth_token = data.split("")[0]
> AttributeError: 'dict' object has no attribute 'split'

@Kpatric! Check you ``def logout_user(data):`` param if it's a dict type.
